chartlets.js/src/lib/components/Plot.tsx

```diff
@@ -2,32 +2,109 @@ import { VegaLite } from "react-vega";
 
 import { type PlotState } from "@/lib/types/state/component";
 import { type ComponentChangeHandler } from "@/lib/types/state/event";
+import {
+  isString,
+  isTopLevelSelectionParameter,
+  type SignalHandler,
+} from "@/lib/types/state/vega";
+import { useCallback, useMemo } from "react";
 
 export interface PlotProps extends Omit<PlotState, "type"> {
   onChange: ComponentChangeHandler;
 }
 
 export function Plot({ id, style, chart, onChange }: PlotProps) {
+  /*
+  Here, we loop through all the params to create map of signals which will be then used to create the map of
+  signal-listeners
+  */
+  const signals: { [key: string]: string } = useMemo(() => {
+    if (!chart) return {};
+    const tempSignals: { [key: string]: string } = {};
+    chart.params?.forEach((param) => {
+      if (isTopLevelSelectionParameter(param)) {
+        if (
+          typeof param.select === "object" &&
+          "on" in param.select &&
+          param.select.on != null
+        ) {
+          const signalName = param.select.on;
+          if (isString(signalName)) {
+            tempSignals[signalName] = param.name;
+          } else {
+            console.warn(
+              "The signal " +
+                param +
+                " is of Stream type (internal Vega-lite type) which is not handled yet.",
+            );
+          }
+        }
+      }
+    });
+    return tempSignals;
+  }, [chart]);
+
+  const handleClickSignal = useCallback(
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    (signalName: string, value: unknown) => {
+      if (id) {
+        return onChange({
+          componentType: "Plot",
+          id: id,
+          property: "points",
```

Comment:
It is really sending `points`? What kind of values do we actually receive? What are the possible property names?

Reply:
This logic has not changed quite for sometime (since Plotly). I will look into it now.

Reply:
The property names depend on how we define the click event on the server side.
Currently, in panel 1, we define it like this
```
click_param = alt.selection_point(
        on="click", name="onClick", fields=["x", variable_name]
    )
```
and the click event output looks like this
![Screenshot from 2024-11-21 16-12-47](https://github.com/user-attachments/assets/6015b5f6-c089-4107-bed0-10299d1cee5c)

So basically, what we send in via server, we get that output as the click event's output.
And since this is a click handling event, the property path can be anything depending on what we want to do. Currently, we just add a new property `points` to the panel object that had the click event.

Reply:
So actually it is up to the user to choose a name for the property that best described the value?

Reply:
Yes, the user can specify more than one property, here `x` and `a` where `a` is the `variable_name`. But our current handler does not do much. It just adds this `points` property to the componentState of this panel. (The properties here should be `x` and `a` instead). This is my understanding.